go/tasks/plugins/webapi/grpc/plugin.go

```diff
@@ -0,0 +1,207 @@
+		State:   res.State,
+		Outputs: res.Outputs,
+	}, nil
+}
+
+func (p Plugin) Delete(ctx context.Context, taskCtx webapi.DeleteContext) error {
+	if taskCtx.ResourceMeta() == nil {
+		return nil
+	}
+	metadata := taskCtx.ResourceMeta().(ResourceMetaWrapper)
+
+	endpoint := getFinalEndpoint(metadata.TaskType, p.cfg.DefaultGrpcEndpoint, p.cfg.EndpointForTaskTypes)
+	client, err := p.getClient(ctx, endpoint, p.connectionCache)
+	if err != nil {
+		return fmt.Errorf("failed to connect external plugin service with error: %v", err)
+	}
+
+	_, err = client.DeleteTask(ctx, &service.TaskDeleteRequest{TaskType: metadata.TaskType, JobId: metadata.JobID})
+	return err
+}
+
+func (p Plugin) Status(ctx context.Context, taskCtx webapi.StatusContext) (phase core.PhaseInfo, err error) {
+	resource := taskCtx.Resource().(*ResourceWrapper)
+	taskInfo := &core.TaskInfo{}
+
+	switch resource.State {
+	case service.State_RUNNING:
+		return core.PhaseInfoRunning(pluginsCore.DefaultPhaseVersion, taskInfo), nil
+	case service.State_PERMANENT_FAILURE:
+		return core.PhaseInfoFailure(pluginErrors.TaskFailedWithError, "failed to run the job", taskInfo), nil
+	case service.State_RETRYABLE_FAILURE:
+		return core.PhaseInfoRetryableFailure(pluginErrors.TaskFailedWithError, "failed to run the job", taskInfo), nil
+	case service.State_SUCCEEDED:
+		if resource.Outputs != nil {
+			err := taskCtx.OutputWriter().Put(ctx, ioutils.NewInMemoryOutputReader(resource.Outputs, nil, nil))
+			if err != nil {
+				return core.PhaseInfoUndefined, err
+			}
+		}
+		return core.PhaseInfoSuccess(taskInfo), nil
+	}
+	return core.PhaseInfoUndefined, pluginErrors.Errorf(pluginsCore.SystemErrorCode, "unknown execution phase [%v].", resource.State)
+}
+
+func getFinalEndpoint(taskType, defaultEndpoint string, endpointForTaskTypes map[string]string) string {
+	if t, exists := endpointForTaskTypes[taskType]; exists {
+		return t
+	}
+
+	return defaultEndpoint
+}
+
+func getClientFunc(ctx context.Context, endpoint string, connectionCache map[string]*grpc.ClientConn) (service.ExternalPluginServiceClient, error) {
+	conn, ok := connectionCache[endpoint]
+	if ok {
+		return service.NewExternalPluginServiceClient(conn), nil
+	}
+	var opts []grpc.DialOption
+	var err error
+
+	opts = append(opts, grpc.WithInsecure())
+	conn, err = grpc.Dial(endpoint, opts...)
+	if err != nil {
+		return nil, err
+	}
+	connectionCache[endpoint] = conn
+	defer func() {
+		if err != nil {
+			if cerr := conn.Close(); cerr != nil {
+				grpclog.Infof("Failed to close conn to %s: %v", endpoint, cerr)
+			}
+			return
+		}
+		go func() {
+			<-ctx.Done()
+			if cerr := conn.Close(); cerr != nil {
+				grpclog.Infof("Failed to close conn to %s: %v", endpoint, cerr)
+			}
+		}()
+	}()
+	return service.NewExternalPluginServiceClient(conn), nil
+}
+
+func newGrpcPlugin() webapi.PluginEntry {
+	supportedTaskTypes := GetConfig().SupportedTaskTypes
+	supportedTaskTypes = append(supportedTaskTypes, "bigquery_query_job_task")
```

Comment:
"bigquery_query_job_task" shouldn't be hardcoded here right?

Reply:
removed it. yes, I think it's better to add it to the configmap in the helm chart.